File: Periodo.java
public class Periodo {
    private int ano;
    private char semestre;
    private String codigo;

    public Periodo(int ano, char semestre) {
        this.ano = ano;
        this.semestre = semestre;
        this.codigo = String.valueOf(ano) + "/" + semestre;
    }

    int obterAno() {
        return ano;
    }

    char obterSemestre() {
        return semestre;
    }

    String obterCodigo() {
        return codigo;
    }

    @Override
    public String toString() {
        return "Periodo: " + codigo;
    }
}

File: cpp_version/Modelo/Periodo.cpp
#include "Periodo.h"

Periodo :: Periodo(int ano, char semestre){
    this->ano = ano;
    this->semestre = semestre;
    this->codigo = to_string(ano) + "/" + semestre;
}

int Periodo :: obeterAno() const{
    return ano;
}

char Periodo :: obterSemestre() const{
    return semestre;
}

string Periodo :: obterCodigo() const{
    return codigo;
}

bool Periodo :: compara(const Periodo* esq, const Periodo* dir){
    return stringCompare(esq->obterCodigo(), dir->obterCodigo());
}

ostream& operator<<(ostream& os, Periodo &periodo){
    os << "Periodo: " << periodo.codigo << endl;
    return os;
}

